File: android/app/src/main/java/com/gwangy/nassspandroidsample/NapSspConfig.kt
package com.gwangy.nassspandroidsample

import android.content.Context

object NapSspConfig {
    private const val DEFAULT_MEDIA_KEY = "10771"
    private val DEFAULT_AD_UNIT_IDS = mapOf(
        "banner_320x100" to "104704",
        "interstitial_320x480_f" to "104703",
        "interstitial_320x480" to "104702",
        "banner_320x50" to "104701",
        "instream_video" to "104591",
        "outstream_video" to "104589",
        "native" to "104588",
        "banner_300x250_f" to "103862",
        "reward_video" to "103722",
        "banner_300x250" to "103485",
    )

    fun mediaKey(context: Context): String =
        AppConfig.getMediaKey(context)?.takeIf { it.isNotBlank() } ?: DEFAULT_MEDIA_KEY

    fun adUnitIds(context: Context): Map<String, String> =
        DEFAULT_AD_UNIT_IDS.mapValues { (key, value) ->
            AppConfig.getAdUnit(context, key)?.takeIf { it.isNotBlank() } ?: value
        }

    val MEDITATION_HINTS = listOf(
        "AdManager" to "io.github.nasmedia-tech:admixer-admanager:1.0.14",
        "AdFit" to "io.github.nasmedia-tech:admixer-adfit:1.0.10",
        "Pangle" to "io.github.nasmedia-tech:admixer-pangle:1.0.10",
        "AppLovin" to "io.github.nasmedia-tech:admixer-applovin:1.0.8",
        "Unity" to "io.github.nasmedia-tech:admixer-unity:1.0.6",
    )
}

File: ios/Sources/NapSspIOSSample/NapSspConfig.swift
import Foundation

enum NapSspConfig {
    private static let defaultMediaKey = "10771"
    private static let defaultAdUnitIDs: [String: String] = [
        "banner_320x100": "104704",
        "interstitial_320x480_f": "104703",
        "interstitial_320x480": "104702",
        "banner_320x50": "104701",
        "instream_video": "104591",
        "outstream_video": "104589",
        "native": "104588",
        "banner_300x250_f": "104703",
        "reward_video": "103722",
        "banner_300x250": "104704"
    ]

    static var mediaKey: String {
        let saved = UserDefaults.standard.string(forKey: "media_key")?.trimmingCharacters(in: .whitespacesAndNewlines)
        return (saved?.isEmpty == false) ? saved! : defaultMediaKey
    }

    static var adUnitIDs: [String: String] {
        var result = defaultAdUnitIDs
        for key in defaultAdUnitIDs.keys {
            if let saved = UserDefaults.standard.string(forKey: "adunit_\(key)")?.trimmingCharacters(in: .whitespacesAndNewlines), !saved.isEmpty {
                result[key] = saved
            }
        }
        return result
    }

    static var mediaKeyInt: Int {
        Int(mediaKey) ?? Int(defaultMediaKey) ?? 0
    }

    static func adUnitID(_ key: String) -> Int {
        Int(adUnitIDs[key] ?? "") ?? Int(defaultAdUnitIDs[key] ?? "") ?? 0
    }

    static var adUnitIDInts: [String: Int] {
        Dictionary(uniqueKeysWithValues: adUnitIDs.map { ($0.key, Int($0.value) ?? 0) })
    }

    static let mediationHints: [(String, String)] = [
        ("GAM", "https://github.com/Nasmedia-Tech/iOS-SSP-GAM-SPM.git"),
        ("AdFit", "https://github.com/Nasmedia-Tech/iOS-SSP-AdFit-SPM.git"),
        ("Pangle", "https://github.com/Nasmedia-Tech/iOS-SSP-Pangle-SPM.git"),
        ("Unity Ads", "https://github.com/Nasmedia-Tech/iOS-SSP-UnityAds-SPM.git"),
        ("AppLovin", "https://github.com/Nasmedia-Tech/iOS-SSP-AppLovin-SPM.git")
    ]
}
